File: html_backup/query_listings.js
const admin = require('firebase-admin');

// Initialize App with application default credentials or local configuration
try {
  admin.initializeApp({
    projectId: 'scalesync-pro'
  });
  const db = admin.firestore();
  db.collection('marketplace_listings').get().then(snapshot => {
    console.log(`Found ${snapshot.size} listings.`);
    snapshot.forEach(doc => {
      console.log(doc.id, JSON.stringify(doc.data(), null, 2));
    });
  }).catch(err => {
    console.error('Error getting documents', err);
  });
} catch (e) {
  console.error(e);
}

File: html_backup/query_listings.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import firestore

try:
    cred = credentials.ApplicationDefault()
    firebase_admin.initialize_app(cred, {
        'projectId': 'scalesync-pro',
    })
    db = firestore.client()
    docs = db.collection('marketplace_listings').stream()
    print("Found listings:")
    for doc in docs:
        print(doc.id, doc.to_dict())
except Exception as e:
    print("Error:", e)
